sonic_platform_base/module_base.py

```diff
@@ -77,6 +89,83 @@ def get_system_eeprom_info(self):
         """
         raise NotImplementedError
 
+    def get_name(self):
+        """
+        Retrieves the name of the module prefixed by CONTROL-CARD, LINE-CARD,
+        FABRIC-CARD
+
+        Returns:
+            string: A string providing the name of the card prefixed by one of the
+            MODULE_TYPE_CONTROL, MODULE_TYPE_LINE, MODULE_TYPE_FABRIC followed by
+            an index.
+            Ex. A Chassis having 1 control-card, 4 line-cards and 6 fabric-cards
+            can provide names CONTROL-CARD1, LINE-CARD1 to LINE-CARD4,
+            FABRIC-CARD1 to FABRIC-CARD6
+        """
+        raise NotImplementedError
+
+    def get_description(self):
+        """
+        Retrieves the platform vendor's product description of the module
+
+        Returns:
+            string: A string providing the product description of the module. This
+            is vendor specific.
+        """
+        raise NotImplementedError
+
+    def get_slot(self):
```

Comment:
what is the difference between this get_slot and the get_my_slot above defined in chassis_base.py?

Reply:
get_slot() is an API in Module class. On the supervisor, in the module_list[], we can loop through and get a slot of each card using get_slot() API on each module object.

get_my_slot() is to get the slot number of where this SONiC instance is running in dis-aggregated chassis i.e. on supervisor0. it might give 1, and line-card0 it might give slot 2.